packages/desktop/components/FormattedBalance.svelte

```diff
@@ -24,15 +27,57 @@
             return { integer: _integer, separator, decimals: '00', unit }
         }
     }
+
+    let textContainerRef: HTMLElement
+    let wrapperRef: HTMLElement
+
+    $: if (textContainerRef && wrapperRef) {
+        void adjustTextType()
+    }
+
+    async function adjustTextType(): Promise<void> {
+        const containerRightEdge = wrapperRef.getBoundingClientRect().right
+
+        // Start with the largest possible font size.
+        let currentIndex = 0
+        usedTextType = textTypeArray[currentIndex]
+
+        // Force an update to the DOM.
+        await tick()
+
+        // Reduce font size until the content fits or until reaching the smallest size.
+        while (currentIndex < textTypeArray.length - 1) {
+            const lastElementOfWrapper = wrapperRef.lastElementChild
+            const lastElementOfTextContainer = textContainerRef.lastElementChild
+
+            if (!lastElementOfWrapper || !lastElementOfTextContainer) return
```

Comment:
Return deserves to be on a separate line, so we can see directly what's happening.
```suggestion
            if (!lastElementOfWrapper || !lastElementOfTextContainer) {
              return
            }
```